nautobot/ipam/models.py

```diff
@@ -574,15 +592,46 @@ def clean(self):
                     {"location": f'Prefixes may not associate to locations of type "{self.location.location_type}".'}
                 )
 
+    def delete(self, *args, **kwargs):
+        """
+        A Prefix with children will be impossible to delete and raise a `ProtectedError`.
+
+        Passing `force_delete=True` will catch the error and explicitly update the
+        `protected_objects` from the exception setting their parent to the old parent of this
+        prefix, and then this prefix will be deleted.
+        """
+        force_delete = kwargs.pop("force_delete", False)
```

Comment:
As raised during the knowledge share today, let's revisit whether this is needed - since deleting a prefix is non-destructive to its descendants and IP addresses, maybe we can just always have this behavior?

Reply:
In the latest patch this is removed entirely and "it just works".